osh/cmd_eval.py

```diff
@@ -1896,7 +1896,13 @@ def ExecuteAndCatch(self, node, cmd_flags=0):
         """
         if cmd_flags & Optimize:
             node = self._RemoveSubshells(node)
-            self._NoForkLast(node)  # turn the last ones into exec
+            # Using _NoForkLast() hides exit code for ERR trap, but only in non-interactive mode!
+            # Example:
+            # - ./bin/osh -c 'trap "echo ERR" ERR ; date X' - does not show ERR
+            # - ./bin/osh -c 'trap "echo ERR" ERR ; date X ; exit' - does show ERR
+            # This is why it is only called interactive mode
+            if self.mutable_opts.Get(option_i.interactive):
```

Comment:
OK I think this is issue 

https://github.com/oilshell/oil/issues/1853

I think we should do this separately, since it's not strictly related to `trap ERR`

I think it should be another option

- `shopt --unset nolastfork` -- not sure if it should be on or off by default - it depends on performance.  there is a Zulip thread about it I will ping
- it should use `exec_opts` not `mutable_opts`